src/py/flwr/client/app.py

```diff
@@ -278,6 +278,7 @@ def _on_sucess(retry_state: RetryState) -> None:
             )
 
     def _on_backoff(retry_state: RetryState) -> None:
+        app_state_tracker.is_connected = False
```

Comment:
There's one small (but rare) issue. If the user starts a ClientApp with `max_tries=1`, the callback `on_backoff` won't be called even if the connection fails. This means the `app_state_tracker.is_connected` will not be set back to `False`. I am not sure if this would impact the `app_state_tracker`'s functionality.

Reply:
@panh99 I just tested it, in this case, as soon as the client loses the connection with the `SuperLink`, it errors out and shuts down properly, so I don't think it's an issue for the `app_state_tracker`

<img width="875" alt="image" src="https://github.com/adap/flower/assets/9378265/32a9c42a-6cf3-48e5-9d84-7a44f5b08ee9">